nexus/src/app/instance.rs

```diff
@@ -1523,6 +1523,38 @@ impl super::Nexus {
         Ok(disk)
     }
 
+    /// Lists affinity groups to which this instance belongs
+    pub(crate) async fn instance_list_affinity_groups(
+        &self,
+        opctx: &OpContext,
+        instance_lookup: &lookup::Instance<'_>,
+        pagparams: &PaginatedBy<'_>,
+    ) -> ListResultVec<db::model::AffinityGroup> {
+        let (.., authz_instance) =
+            instance_lookup.lookup_for(authz::Action::ListChildren).await?;
+        self.db_datastore
+            .instance_list_affinity_groups(opctx, &authz_instance, pagparams)
+            .await
+    }
+
+    /// Lists anti-affinity groups to which this instance belongs
+    pub(crate) async fn instance_list_anti_affinity_groups(
+        &self,
+        opctx: &OpContext,
+        instance_lookup: &lookup::Instance<'_>,
+        pagparams: &PaginatedBy<'_>,
+    ) -> ListResultVec<db::model::AntiAffinityGroup> {
+        let (.., authz_instance) =
+            instance_lookup.lookup_for(authz::Action::ListChildren).await?;
```

Comment:
Should this (also or instead) check list children on projects since the groups are children of the project? I think that check is implied by this check because anyone with viewer+ on the project has list children on the project and the instance, but that's sort of incidental and might be better if this check was more direct.

Reply:
I see we do the same for listing instance disks, which are directly analogous from an authz POV. Seems fine?

https://github.com/oxidecomputer/omicron/blob/0c92213cbf480848862926873d1c8e633102471e/nexus/src/app/instance.rs?plain=1#L1410-L1422